Src/GoogleApis.Core/Apis/Http/ConfigurableMessageHandler.cs

```diff
@@ -64,22 +74,115 @@ public class ConfigurableMessageHandler : DelegatingHandler
         private readonly IList<IHttpExecuteInterceptor> executeInterceptors =
             new List<IHttpExecuteInterceptor>();
 
-        /// <summary>Gets a list of <see cref="IHttpUnsuccessfulResponseHandler"/>.</summary>
-        public IList<IHttpUnsuccessfulResponseHandler> UnsuccessfulResponseHandlers
+        /// <summary>
+        /// Gets a copied list of <see cref="IHttpUnsuccessfulResponseHandler"/>.
+        /// <remarks>
+        /// Since version 1.10, the return value had been changed from <c>IList</c> to <c>IEnumerable</c> in order to
+        /// keep this class thread-safe. The returned list is also a copied version of the handlers. More info is
+        /// available on <a href="https://github.com/google/google-api-dotnet-client/issues/592">Issue 592</a>.
+        /// </remarks>
+        /// </summary>
+        public IEnumerable<IHttpUnsuccessfulResponseHandler> UnsuccessfulResponseHandlers
+        {
+            get
+            {
+                lock (unsuccessfulResponseHandlersLock)

```

Comment:
2. I think that those were introduced in .NET 4.5
https://visualstudiomagazine.com/articles/2012/08/07/new-read-only-collection-interfaces-for-net.aspx

We can use this one - https://msdn.microsoft.com/en-us/library/ms132474(v=vs.100).aspx, but it's NOT an interface.

I think I'm leaning toward ReadOnlyCollection<T>, but again I don't mind using IEnumerable.
This one: http://stackoverflow.com/questions/491375/readonlycollection-or-ienumerable-for-exposing-member-collections outlines that ReadOnlyCollection is better for our case, since users won't be able to cast and add...

So bottom line: ReadOnlyCollection (class and not interface), right?